packages/sync-server/src/app-pluggyai/app-pluggyai.js

```diff
@@ -0,0 +1,221 @@
+          balanceType: 'expected',
+          referenceDate: date,
+        },
+        {
+          balanceAmount: {
+            amount: startingBalance,
+            currency: account.currencyCode,
+          },
+          balanceType: 'interimAvailable',
+          referenceDate: date,
+        },
+      ];
+
+      const all = [];
+      const booked = [];
+      const pending = [];
+
+      for (const trans of transactions) {
+        const newTrans = {};
+
+        let dateToUse = 0;
+
+        if (trans.status === 'PENDING') {
+          newTrans.booked = false;
+        } else {
+          newTrans.booked = true;
+        }
+        dateToUse = trans.date;
+
+        const transactionDate = new Date(dateToUse);
+
+        if (transactionDate < startDate) {
+          continue;
+        }
+
+        newTrans.date = getDate(transactionDate);
+
+        newTrans.payeeName = '';
+        if (
+          trans.merchant &&
+          (trans.merchant.name || trans.merchant.businessName)
+        ) {
+          newTrans.payeeName =
+            trans.merchant.name || trans.merchant.businessName;
+        } else if (
+          trans.type === 'DEBIT' &&
+          trans.paymentData &&
+          trans.paymentData.receiver &&
+          trans.paymentData.receiver.name
+        ) {
+          newTrans.payeeName = trans.paymentData.receiver.name;
+        } else if (
+          trans.type === 'CREDIT' &&
+          trans.paymentData &&
+          trans.paymentData.payer &&
+          trans.paymentData.payer.name
+        ) {
+          newTrans.payeeName = trans.paymentData.payer.name;
+        } else if (
+          trans.type === 'DEBIT' &&
+          trans.paymentData &&
+          trans.paymentData.receiver &&
+          trans.paymentData.receiver.documentNumber &&
+          trans.paymentData.receiver.documentNumber.value
+        ) {
+          newTrans.payeeName = trans.paymentData.receiver.documentNumber.value;
+        } else if (
+          trans.type === 'CREDIT' &&
+          trans.paymentData &&
+          trans.paymentData.payer &&
+          trans.paymentData.payer.documentNumber &&
+          trans.paymentData.payer.documentNumber.value
+        ) {
+          newTrans.payeeName = trans.paymentData.payer.documentNumber.value;
+        }
+
+        newTrans.remittanceInformationUnstructured = trans.descriptionRaw;
+        let amountInCurrency = trans.amountInAccountCurrency ?? trans.amount;
+        amountInCurrency = Math.round(amountInCurrency * 100) / 100;
+
+        newTrans.transactionAmount = {
+          amount:
+            account.type === 'BANK' ? amountInCurrency : -amountInCurrency,
+          currency: trans.currencyCode,
+        };
+        newTrans.transactionId = trans.id;
+        newTrans.valueDate = newTrans.bookingDate; //always undefined?
+
+        if (newTrans.booked) {
+          booked.push(newTrans);
+        } else {
+          pending.push(newTrans);
+        }
+        all.push(newTrans);
+      }
```

Comment:
It would be nice to return all of the transaction properties here so that the custom mapping in the new bank sync page has something to look at.

```suggestion
        const finalTrans = { ...trans, ...newTrans };
        if (newTrans.booked) {
          booked.push(finalTrans);
        } else {
          pending.push(finalTrans);
        }
        all.push(finalTrans);
      }
```

Reply:
done